activesupport/test/message_encryptor_test.rb

```diff
@@ -106,8 +106,118 @@ def test_messing_with_aead_values_causes_failures
     assert_aead_not_decrypted(encryptor, [text, iv, auth_tag[0..-2]] * "--")
   end
 
-  private
+  def test_metadata_purpose
+    encryptor = ActiveSupport::MessageEncryptor.new(@secret, cipher: "aes-256-gcm")
+    encrypted_message = encryptor.encrypt_and_sign("foo", purpose: "bar")
+    assert_equal "foo", encryptor.decrypt_and_verify(encrypted_message, purpose: "bar")
+    assert_nil encryptor.decrypt_and_verify(encrypted_message, purpose: "A completely different purpose")
```

Comment:
While possible, `"A complete different purpose"` isn't really indicative of how people should use the purpose. It should be an identifier — think `:login`, `:authentication_token` (we should support symbols too).